src/app/extensions/quickorder/shared/direct-order/direct-order.component.ts

```diff
@@ -44,7 +41,13 @@ export class DirectOrderComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   ngAfterViewInit() {
-    this.context.connect('sku', this.directOrderForm.get('sku').valueChanges);
+    this.context.connect(
+      'sku',
+      this.directOrderForm.get('sku').valueChanges.pipe(
+        tap(() => (this.loading = true)),
+        debounceTime(500)
```

Comment:
double requests? Did you check out `distinctUntilChanged`?

Reply:
One Request came from the context. The other from the check of the asnyc validator, which asked the shopping facade. Same issue is in the quickorder page. I actually try to fix this there too.